modules/vmfcore/java/Metadata.java

```diff
@@ -0,0 +1,153 @@
+
+public class Metadata
+{
+	static
+    {
+        System.loadLibrary("vmf");
+    }
+	
+	protected long nativeObj;
+	
+	public static final long UNDEFINED_DURATION = 0;
+	
+	public static final long NUMOFFRAMES_DEFAULT = 1;
+	
+	protected Metadata (long addr)
+    {
+        if (addr == 0)
+            throw new java.lang.UnsupportedOperationException("Native object address is NULL");
+        
+        nativeObj = addr;
+    }
+	
+	public Metadata (MetadataDesc mdDescription)
+	{
+		nativeObj = n_Metadata (mdDescription.nativeObj);
+	}
+	
+	public void copy (Metadata metadata)
+	{
+		if ((nativeObj == 0) || (metadata.nativeObj == 0))
```

Comment:
it's better to block the possibility of having `nativeObj == 0` than checking this everywhere...